velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -260,5 +260,28 @@ TEST_F(DateTimeFunctionsTest, dateAdd) {
   EXPECT_EQ(parseDate("1969-12-31"), dateAdd(parseDate("5881580-07-11"), kMin));
 }
 
+TEST_F(DateTimeFunctionsTest, dateDiffDate) {
+  const auto dateDiff = [&](std::optional<int32_t> date1,
+                            std::optional<int32_t> date2) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_diff(c0, c1)", {date1, date2}, {DATE(), DATE()});
+  };
+
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateDiff(1, std::nullopt));
+  EXPECT_EQ(std::nullopt, dateDiff(std::nullopt, 0));
+
+  // Simple tests
```

Comment:
nit: out of completeness, could add a date minus itself to ensure it returns 0.

Reply:
Sorry for late reply. Just fixed it.